packages/mnemonic-seed-manager/src/agent/MnemonicSeedManager.ts

```diff
@@ -113,4 +118,49 @@ export class MnemonicInfoGenerator implements IAgentPlugin {
       .orWhere('hash = :hash', { hash: args.hash })
       .execute()
   }
+
+  private async saveMasterKey(args: IMnemonicInfoStoreArgs): Promise<UpdateResult> {
+    if (args.masterKey) {
+      return (await this.dbConnection)
+        .createQueryBuilder()
+        .update(MnemonicInfo)
+        .set({ masterKey: args.masterKey })
+        .where('id = :id', { id: args.id })
+        .orWhere('hash = :hash', { hash: args.hash })
+        .execute()
+    }
+    throw new Error('Master Key needs to be provided.')
+  }
+
+  private async generateMasterKey(args: IMnemonicInfoStoreArgs): Promise<IMnemonicInfoKeyResult> {
+    const mnemonic = (await this.getMnemonicInfo({ id: args.id, hash: args.hash })).mnemonic
+    if (mnemonic) {
+      const seed = await bip39.mnemonicToSeed(mnemonic?.join(' '))
```

Comment:
Still not using our own generate seed method, which does the same as this line, including the to hex conversion you are doing later on the seed

Reply:
That's the call to the library to generate the master key.

Reply:
? 
`await bip39.mnemonicToSeed(mnemonic?.join(' '))`

and later

`seed.toString('hex')`


is not our `generateSeed`

Reply:
It's not ready, the ticket still assigned to me. ;-)

Reply:
Fixed!